src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -341,6 +346,15 @@ DistOpsValidityState(Node *node, const DistributeObjectOps *ops)
 		bool isPostprocess = false;
 		List *objectAddresses = ops->address(node, missingOk, isPostprocess);
 
+		if (HasDropCommandViolatesOwnership(node))
+		{
+			/*
+			 * found object with an invalid ownership, PG will complain if there is any object
+			 * with an invalid ownership.
+			 */
```

Comment:
We need a comment on why we are only interested in ownership checks for `DROP` commands.

Please add something like below (in a better format), maybe as a function comment:

/*
  Currently there is only one pg vanilla tests where tests fails due to object ownership. The command that is failing is `DROP`. If in the future we hit other commands like this, we should expand this function.

*/

Reply:
I agree about adding a GUC enabled by default and clients can switch it off. I plan to add it on the main PR #6018.